openvdb/Grid.h

```diff
@@ -1355,7 +1355,10 @@ template<typename TreeT>
 inline void
 Grid<TreeT>::pruneGrid(float tolerance)
 {
-    this->tree().prune(ValueType(zeroVal<ValueType>() + tolerance));
+    constexpr bool IsString = std::is_same<ValueType, std::string>::value;
+    using ZeroValType =
+        typename std::conditional<IsString, std::string, float>::type;
+    this->tree().prune(ValueType(zeroVal<ZeroValType>() + tolerance));
```

Comment:
@pcucka - do you have any thoughts on this? I had a look at fixing this one a while back but didn't get too far. It feels like the tolerance should only really apply to floating-point value types?

Reply:
No, the tolerance is applicable to all numeric types, both scalar and vector, and probably to some user-defined types as well.

Reply:
What about this:

```
    constexpr bool Convertible = std::is_convertible<float, ValueType>::value;
    using ToleranceType =
        typename std::conditional<Convertible, ValueType, float>::type;

    this->tree().prune(ValueType(zeroVal<ValueType>() + ToleranceType(tolerance)));
```

Reply:
What warning(s) were you getting with the original code?

Reply:
Conversion warnings from -Wconversion - the expression is wrapped in a cast but the addition is left implicit. This was most of the warnings in this PR, hence the precision concerns: 

```
Grid.h:1361:52: warning: conversion to ‘float’ from ‘long int’ may alter its value [-Wconversion]
     this->tree().prune(ValueType(zeroVal<ValueType>() + tolerance));
```

Reply:
In that case, I think you're on the right track, but I'm not sure your solution would address the case where `ValueType` is, say, `Vec3i`.  Also, if `ValueType` _is_ convertible, then there's no need to add `zeroVal`.

Reply:
Yeah I don't think is_convertible is returning true for something like Vec3i, but there's never any warnings with the expression for types like Vec3i which have a templated + operator (which forwards to += where the casting is static). The warnings I got from this method were from scalar + float where scalar wasn't a float, resulting in the implicit cast. 

I left add zeroVal in for when ValueType is convertible as introducing a runtime branch would bring back the warnings again.

Reply:
I've ended up with the following:
```
    constexpr bool Constructible =
        openvdb::CanConvertType<float, ValueType>::value;
    using ToleranceType =
        typename std::conditional<Constructible, ValueType, float>::type;
```

Reply:
Unfortunately, that seems to have reintroduced the warning:
https://travis-ci.org/AcademySoftwareFoundation/openvdb/jobs/508949989#L1007

Reply:
Well that's annoying, thought I could get away with using `is_constructible` but didn't see clang fail - i'll change it back to `is_convertible`